Extensions/StyleCop/StyleCopTask/StyleCop.ps1

```diff
@@ -46,11 +47,23 @@ Add-Content $summaryMdPath ("`nStyleCop found [{0}] violations across [{1}] proj
 Write-verbose "Uploading summary results file"
 Write-Host "##vso[build.uploadsummary]$summaryMdPath"
 
+# Set the message that will be returned
 if ($result.OverallSuccess -eq $false)
 {
-   Write-Error ("StyleCop found [{0}] violations across [{1}] projects" -f $result.TotalViolations, $result.ProjectsScanned)
+   $resultMessage = ("StyleCop found [{0}] violations across [{1}] projects" -f $result.TotalViolations, $result.ProjectsScanned)
 } 
 else
 {
-   Write-Verbose ("StyleCop found [{0}] violations warnings across [{1}] projects" -f $result.TotalViolations, $result.ProjectsScanned)
+   $resultMessage = ("StyleCop found [{0}] violations warnings across [{1}] projects" -f $result.TotalViolations, $result.ProjectsScanned)
 }
+
+# Determine if the task should error
+if ([int]$result.TotalViolations -gt [int]$allowableViolationCount)
```

Comment:
My thought here was I called the setting 'allowable' so I'm allowing 10 failures, the task should only fail if there are more than the allowable number.